pkg/impersonation/impersonation.go

```diff
@@ -214,6 +234,7 @@ func (i *Impersonator) rulesForUser() []rbacv1.PolicyRule {
 	}}
 
 	if groups := i.user.GetGroups(); len(groups) > 0 {
+		sort.Strings(groups)
```

Comment:
I think it makes sense to add comments to all these sorts added to explain why they are needed. To someone not familiar with the reason it may seem we do it just so it looks nicer or something trivial.

Reply:
done